ste/sender-blobFolders.go

```diff
@@ -53,6 +54,53 @@ func newBlobFolderSender(jptm IJobPartTransferMgr, destination string, p pipelin
 	return out, nil
 }
 
+func (b *blobFolderSender) overwriteDFSProperties() (string, error) {
+	b.jptm.Log(pipeline.LogWarning, "It is impossible to completely overwrite a folder with existing content under it on a hierarchical namespace storage account. A best-effort attempt will be made, but if CPK does not match the transfer will fail.")
+
+	b.metadataToApply["hdi_isfolder"] = "true" // Set folder metadata flag
```

Comment:
Is this function generic as name does not suggest this is only for folders though we are setting hdi_isfolder unconditionally here.

Reply:
@vibhansa-msft The type is a blobFolderSender so I think its safe to always add the metadata